packages/codemod/bin/vibe-codemod.ts

```diff
@@ -0,0 +1,253 @@
+    process.exit(1);
+  }
+
+  function isGitClean() {
+    try {
+      const status = spawnSync("git", ["status", "--porcelain"], { encoding: "utf-8" }).stdout.trim();
+      return status === "";
+    } catch (error) {
+      console.error(chalk.red("Error checking Git status:"));
+      process.exit(1);
+    }
+  }
+
+  if (!isGitClean()) {
+    console.warn(
+      chalk.yellow(
+        "Warning: Your Git working directory is not clean. It is recommended to only run this script with a clean working directory."
+      )
+    );
+
+    const proceed = readlineSync.question("Do you want to proceed anyway? (y/N) ");
+    if (proceed.toLowerCase() !== "y") {
+      console.log("Operation cancelled.");
+      process.exit(1);
+    }
+  }
+
+  const verbosityLevel: number = verbose ? 2 : 0;
+  const outputTarget: string = verbose ? `>> "${logFile}"` : "";
+
+  let successCount = 0;
+  let failureCount = 0;
+  let errorsCount = 0;
+  const startTime = Date.now();
+
+  console.log(chalk.green(`Running transformations for migration: ${migrationType} in directory: ${targetDir}`));
+
+  const spinner = ora(`Running transformations...`).start();
+
+  async function runSingleTransformation(transform: string) {
+    return new Promise(resolve => {
+      const child = spawn(
+        "node",
+        [
+          require.resolve("jscodeshift/bin/jscodeshift"),
+          `--extensions=${extensions.join(",")}`,
+          "--ignore-pattern=node_modules",
+          "--ignore-pattern=**/*.d.ts",
+          `--verbose=${verbosityLevel}`,
+          "--max-workers=8",
+          "--no-babel",
+          "-t",
+          transform,
+          targetDir,
+          outputTarget
+        ],
+        {
+          stdio: verbose ? "inherit" : ["inherit", "pipe", "pipe"],
+          shell: true,
+          env: { ...process.env, FORCE_COLOR: "1" }
+        }
+      );
+
+      let hasError = false;
+
+      if (child.stderr) {
+        child.stderr.on("data", data => {
+          const errorOutput = data.toString();
+          process.stderr.write(errorOutput);
+          if (errorOutput.includes("ERROR")) {
+            hasError = true;
+            errorsCount++;
+          }
+        });
+      }
+
+      if (child.stdout) {
+        child.stdout.on("data", data => {
+          const output = data.toString();
+          if (output.includes("ERROR")) {
+            hasError = true;
+            errorsCount++;
+          }
+        });
+      }
+
+      child.on("exit", code => {
+        if (code !== 0 || hasError) {
+          resolve(false);
+        } else {
+          resolve(true);
+        }
+      });
+    });
+  }
+
+  const transformationFiles: string[] = globby.sync(`${transformationsDir}/*.js`, {
+    ignore: ["node_modules/**", "**/*.d.ts"]
+  });
+
+  if (transformationFiles.length === 0) {
+    console.error(chalk.red("No transformation files found. Please check the directory path."));
```

Comment:
maybe "please check the migration name supplied?"
as this scenario shouldn't happen unless specified with wrong migration name?